.ci/diff-report.sh

```diff
@@ -39,8 +39,8 @@ download-files)
   mkdir .ci-temp
   echo "Downloading files..."
 
-  # check for projects link from PR, if not found use default from contribution repo
-  LINK="${LINK_FROM_PR:-$DEFAULT_PROJECTS_LINK}"
+  # Use DIFF_PROJECT_LINK if set, otherwise fall back to LINK_FROM_PR or DEFAULT_PROJECTS_LINK
+  LINK="${DIFF_PROJECT_LINK:-${LINK_FROM_PR:-$DEFAULT_PROJECTS_LINK}}"
```

Comment:
not ok.

Lets create separate PR and just refactor/rename LINK_FROM_PR to  DIFF_PROJECTS_LINK.
before execution of this shell scripts, github action will do 
`DIFF_PROJECTS_LINK: ${{ needs.parse_body.outputs.projects_link }}`
at
https://github.com/checkstyle/checkstyle/blob/6d8f8f2f9064ab7706fd5c1dd5c715fe0b4ac599/.github/workflows/diff-report.yml#L117-L120

after  merge of such simple refactoring
new action will do `DIFF_PROJECTS_LINK: ${{ needs.parse_comment.outputs.projects_link }}` before action execution.
So no updates in existing will be required.

Reply:
https://github.com/checkstyle/checkstyle/issues/15261 is merged, please rebase, and lets see what else is problem.